test/test_testing.py

```diff
@@ -2699,6 +2700,99 @@ def test_container_pebble_ready(self):
         )
 
 
+class TestNetwork(unittest.TestCase):
+    def test_add_network_defaults(self):
+        harness = Harness(CharmBase)
+        self.addCleanup(harness.cleanup)
+
+        harness.add_network('db', '10.0.0.10')
+
+        binding = harness.model.get_binding('db')
+        self.assertEqual(binding.name, 'db')
+        network = binding.network
+        self.assertEqual(network.bind_address, ipaddress.IPv4Address('10.0.0.10'))
+        self.assertEqual(network.ingress_address, ipaddress.IPv4Address('10.0.0.10'))
+        self.assertEqual(network.ingress_addresses, [ipaddress.IPv4Address('10.0.0.10')])
+        self.assertEqual(network.egress_subnets, [ipaddress.IPv4Network('10.0.0.0/24')])
+        self.assertEqual(len(network.interfaces), 1)
+        interface = network.interfaces[0]
+        self.assertEqual(interface.name, 'eth0')
+        self.assertEqual(interface.address, ipaddress.IPv4Address('10.0.0.10'))
+        self.assertEqual(interface.subnet, ipaddress.IPv4Network('10.0.0.0/24'))
+
+    def test_add_network_all_args(self):
+        harness = Harness(CharmBase)
+        self.addCleanup(harness.cleanup)
+
+        relation_id = harness.add_relation('db', 'postgresql')
+        harness.add_network('db', '10.0.0.10',
+                            relation_id=relation_id,
+                            cidr='10.0.0.0/8',
+                            interface='eth1',
+                            ingress_addresses=['10.0.0.1', '10.0.0.2'],
+                            egress_subnets=['10.0.0.0/8', '10.0.0.0/16'])
```

Comment:
The nittiest of nits: this confused me a little later in the tests because `10.0.0.0/16` is a subset of `10.0.0.0/8`. Might be worth using `10.10.0.0/16` and `10.0.0.0/8` or something obviously different just for readability.

Reply:
Good call -- fixed.